src/config/environmentConfig.ts

```diff
@@ -0,0 +1,49 @@
+'use strict';
+
+/**
+ * Validates and sets the environment configuration based on the provided schema.
+ * If the environment config is invalid, an error is thrown.
+ *
+ * @throws {Error} If the environment config is invalid
+ *
+ * @returns {EnvironmentConfig} The validated environment configuration
+ */
+
+import * as joi from '@hapi/joi';
+import { EnvironmentConfig } from '../models/environmentConfig.interface';
+
+const envs = ['local', 'development', 'qa', 'production', 'test'];
```

Comment:
QA & Test are not the same? 
local, development, test, pre production & production looks good to me. please confirm it with Oliver.

Reply:
During the execution of test cases, the environment is referred by jest as `tes`. I gave the name `qa` to the actual test environment. We are capable of updating that. Once the deployment environments are ready, we can update this since we don't have much clarity.